components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -3744,24 +3744,56 @@ def get_match(name):
                 templateBody.append(arrayObject)
             conversions.append(CGIfWrapper("value.get().is_object()", templateBody))
 
-        otherMemberTypes = [
-            t for t in memberTypes if t.isPrimitive() or t.isString() or t.isEnum()
-        ]
-        if len(otherMemberTypes) > 0:
-            assert len(otherMemberTypes) == 1
-            memberType = otherMemberTypes[0]
-            if memberType.isEnum():
-                name = memberType.inner.identifier.name
+        stringTypes = [t for t in memberTypes if t.isString() or t.isEnum()]
+        numericTypes = [t for t in memberTypes if t.isNumeric()]
+        booleanTypes = [t for t in memberTypes if t.isBoolean()]
+        if stringTypes or numericTypes or booleanTypes:
+            assert len(stringTypes) <= 1
+            assert len(numericTypes) <= 1
+            assert len(booleanTypes) <= 1
+
+            def getStringOrPrimitiveConversion(memberType):
+                typename = get_name(memberType)
+                return CGGeneric(get_match(typename))
+            other = []
+            stringConversion = map(getStringOrPrimitiveConversion, stringTypes)
+            numericConversion = map(getStringOrPrimitiveConversion, numericTypes)
+            booleanConversion = map(getStringOrPrimitiveConversion, booleanTypes)
+            if stringConversion:
+                if booleanConversion:
+                    other.append(CGIfWrapper("value.get().is_boolean()", booleanConversion[0]))
+                if numericConversion:
+                    other.append(CGIfWrapper("value.get().is_number()", numericConversion[0]))
+                other.append(stringConversion[0])
+            elif numericConversion:
+                if booleanConversion:
+                    other.append(CGIfWrapper("value.get().is_boolean()", booleanConversion[0]))
+                other.append(numericConversion[0])
             else:
-                name = memberType.name
-            match = (
-                "match %s::TryConvertTo%s(cx, value) {\n"
-                "    Err(_) => return Err(()),\n"
-                "    Ok(Some(value)) => return Ok(%s::%s(value)),\n"
-                "    Ok(None) => (),\n"
-                "}\n") % (self.type, name, self.type, name)
-            conversions.append(CGGeneric(match))
-            names.append(name)
+                assert booleanConversion
+                other.append(booleanConversion[0])
+            if hasObjectTypes:
+                if object:
```

Comment:
We don't support `object` being non-null here; just drop the conditionals and use
```python
conversions.append(CGList(other, "\n\n"))
```